pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -261,6 +262,45 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
 	}
 
+	poolAndRadosNamespaceName := fmt.Sprintf("%s/%s", pool, getRadosNamespaceName(cephBlockPoolRadosNamespace))
+	if cephBlockPoolRadosNamespace.Spec.Mirroring.Enabled {
+		err = cephclient.EnableRBDRadosNamespaceMirroring(r.context, r.clusterInfo, poolAndRadosNamespaceName, cephBlockPoolRadosNamespace.Spec.Mirroring.RemoteNamespace, string(cephBlockPoolRadosNamespace.Spec.Mirroring.Mode))
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to enable rbd rados namespace mirroring")
+		}
+
+		// Schedule snapshots
+		if len(cephBlockPoolRadosNamespace.Spec.Mirroring.SnapshotSchedules) > 0 {
```

Comment:
I have concerns about how this conditional interacts with the `EnableSnapshotSchedules()` call to `removeSnapshotSchedules()`. It seems to me that if a user removes the last snapshot schedule, Rook might not remove it.

Generally, operators can avoid many bugs by always processing logic regardless of whether lists have items or not.

Reply:
Make sense, We have all over in the code `	if c.poolSpec.Mirroring.SnapshotSchedulesEnabled() {`

should we refactor it?